arbor/simulation.cpp

```diff
@@ -558,10 +565,47 @@ void simulation::set_local_spike_callback(spike_export_function export_callback)
     impl_->local_export_callback_ = std::move(export_callback);
 }
 
+void simulation::set_epoch_callback(epoch_function epoch_callback, bool global) {
```

Comment:
`global` is redundant? Or is this compatibility with `spike_callback`?

Reply:
Good point. That is a hang over from an earlier draft of the epoch callback.
I realised that the `global` information is redundant, and the caller can decide whether to attach a callback on one or more distributed contexts for themselves, e.g.:
https://github.com/arbor-sim/arbor/pull/1873/files#diff-5827156d5c4d2c924b0def9bbf4e51db85e1ea4a0c45c75328ca38dab6ea8e12R190